python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -344,3 +351,47 @@ def _reuse_task_definition(self, task_definition, metadata, image, secrets):
 
     def _task_metadata(self):
         return default_ecs_task_metadata(self.ec2, self.ecs)
+
+    @property
+    def supports_check_run_worker_health(self):
+        return True
+
+    def check_run_worker_health(self, run: PipelineRun):
+
+        tags = self._get_run_tags(run.run_id)
+
+        if not (tags.arn and tags.cluster):
+            return CheckRunHealthResult(WorkerStatus.UNKNOWN, "")
+
+        tasks = self.ecs.describe_tasks(tasks=[tags.arn], cluster=tags.cluster).get("tasks")
+        if not tasks:
+            return CheckRunHealthResult(WorkerStatus.UNKNOWN, "")
+
+        t = tasks[0]
+
+        running_statuses = ["PROVISIONING", "PENDING", "ACTIVATING", "RUNNING"]
+        stopped_statuses = ["DEACTIVATING", "STOPPING", "DEPROVISIONING", "STOPPED"]
```

Comment:
I wonder if we need some additional polling here - like if we catch a task while it's `DEACTIVATING`, `STOPPING`, or `DEPROVISIONING`, if we should still count it as `WorkerStatus.RUNNING`? My fear is that ECS might not yet have populated keys like `exitCode` or `stoppedReason` until it actually reaches a status of `STOPPED`.

Reply:
yeah, i moved everything but Stopped into the running statuses list, hopefully after seeing this in action we can decide if some should be moved to stopped